build/build.js

```diff
@@ -258,6 +264,43 @@ class Build {
     }
 
     /**
+     * Iterates over all languages in the database and adds a list of countries
+     * where the given language is the or an official language
+     * @param {object} json - The JSON object
+     * @return {object}
+     */
+    addOfficialLang(json) {
+        let clone = JSON.parse(JSON.stringify(json));
+        const languages = officialLang.supplemental.languageData,
+            territories = territoryInfo.supplemental.territoryInfo;
+        Object.keys(json).forEach(lang => {
+            Object.keys(json[lang]).forEach(variant => {
+                const meta = clone[lang][variant].metadata;
+                let official = [];
+                // languageData contains ALL territories where a language is
+                // spoken, so we will cross-reference with the territoryInfo
+                // to only find offical languages
+                if(languages[variant]["_territories"]) {
+                    languages[variant]["_territories"].forEach(trty => {
+                        const vrnt =
+                            territories[trty].languagePopulation[variant];
+                        // official or official_regional language if defined
+                        if(vrnt["_officialStatus"]) {
+                            official.push(trty);
+                        }
+                    });
+                }
+                if(meta.official && Array.isArray(meta.official)) {
+                    meta.official.concat(official);
```

Comment:
This can't work. `Array.prototype.concat` just returns the merged array, it doesn't modify the original array by reference.